src/device_info.sh

```diff
@@ -341,6 +345,69 @@ function get_desktop_environment()
   device_info_data['desktop_environment']="$formatted_de"
 }
 
+# This function populates kernel variables from the device_info_data
+# variable.
+#
+# @target Target machine
+function get_kernel_info()
+{
+  local target="$1"
+  local flag="$2"
+  local cmd_name
+  local cmd_version
+  local cmd_release
+  local cmd_machine
+  local kernel_name
+  local kernel_version
+  local kernel_release
+  local kernel_machine_type
+
+  cmd_name="uname --kernel-name"
+  cmd_release="uname --kernel-release"
+  cmd_version="uname --kernel-version"
+  cmd_machine="uname --machine"
+
+  case "$target" in
+    2) # LOCAL_TARGET
+      show_verbose "$flag" "$cmd_name"
+      kernel_name=$(cmd_manager 'SILENT' "$cmd_name")
+
+      show_verbose "$flag" "$cmd_release"
+      kernel_release=$(cmd_manager 'SILENT' "$cmd_release")
+
+      show_verbose "$flag" "$cmd_version"
+      kernel_version=$(cmd_manager 'SILENT' "$cmd_version")
+
+      show_verbose "$flag" "$cmd_machine"
+      kernel_machine=$(cmd_manager 'SILENT' "$cmd_machine")
+      ;;
+    3) # REMOTE_TARGET
+      show_verbose "$flag" "$cmd_name"
+      kernel_name=$(cmd_remotely 'SILENT' "$cmd_name")
+
+      show_verbose "$flag" "$cmd_release"
+      kernel_release=$(cmd_remotely 'SILENT' "$cmd_release")
+
+      show_verbose "$flag" "$cmd_version"
+      kernel_version=$(cmd_remotely 'SILENT' "$cmd_version")
+
+      show_verbose "$flag" "$cmd_machine"
+      kernel_machine=$(cmd_remotely 'SILENT' "$cmd_machine")
+      ;;
+  esac
+
+  if [[ "$flag" == 'TEST_MODE' ]]; then
+    printf '%s\n%s\n' "$cmd_name" "$cmd_release" \
+      "$cmd_version" "$cmd_machine"
```

Comment:
Use a single line.

Reply:
You need to make this change in this commit (first).